spec/services/search/buffer_suggestions_builder_spec.rb

```diff
@@ -0,0 +1,59 @@
+require "rails_helper"
+include DistanceHelper
+
+RSpec.describe Search::BufferSuggestionsBuilder do
+  let(:subject) { described_class.new(location, search_params) }
+
+  describe "#get_buffers_suggestions" do
+    let(:search_params) do
+      {
+        hits_per_page: 10,
+      }
+    end
+    let(:location) { "Tower Hamlets" }
+    let(:buffer_coordinates) { [5.004562496029994090, 56.50833566307333, 5.005710530794815389, 56.5051084208278] }
+    let(:buffer_hash) do
+      {
+        "5" => buffer_coordinates,
+        "10" => buffer_coordinates,
+        "15" => buffer_coordinates,
+        "20" => buffer_coordinates,
+        "25" => buffer_coordinates,
+      }
+    end
+
+    before do
+      allow(LocationPolygon).to receive_message_chain(:with_name, :buffers).and_return(buffer_hash)
+    end
+
+    context "when there is only one vacancy in any of the buffer polygons" do
+      before do
+        allow_any_instance_of(Search::AlgoliaSearchRequest).to receive_message_chain(:stats, :last).and_return(1)
+      end
+
+      it "suggests the buffer with the smallest radius" do
+        expect(subject.buffer_suggestions).to eql([["5", 1]])
+      end
+    end
+
+    context "when there are vacancies in the wider buffer polygons not found in the smaller polygons" do
+      before do
+        allow(Search::AlgoliaSearchRequest).to receive_message_chain(:new, :stats, :last).and_return(1, 2, 3, 4, 5)
```

Comment:
Really cool, didn't know you could do this. But I feel like the approach in `radius_suggestions_builder_spec.rb` might be more valuable.

```ruby
mock_algolia_search(
  vacancies_1, 1, "maths",
  arguments_for_algolia.merge(aroundRadius: Search::LocationBuilder.convert_radius_in_miles_to_metres(5))
)
```

for example is testing the correct parameters are called as well, which I think we lose out on here otherwise